static/app/views/performance/newTraceDetails/traceAnalytics.tsx

```diff
@@ -1,19 +1,38 @@
 import * as Sentry from '@sentry/react';
 
 import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
 import {trackAnalytics} from 'sentry/utils/analytics';
 import type {TraceTree} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
 import type {TraceType} from 'sentry/views/performance/newTraceDetails/traceType';
 
-const trackTraceMetadata = (tree: TraceTree, organization: Organization) => {
+const trackTraceMetadata = (
+  tree: TraceTree,
+  projects: Project[],
+  organization: Organization
+) => {
   Sentry.metrics.increment(`trace.trace_shape.${tree.shape}`);
+
   // space[1] represents the node duration (in milliseconds)
   const trace_duration_seconds = (tree.root.space?.[1] ?? 0) / 1000;
+  const projectSlugs = [
+    ...new Set(
+      tree.list.map(node => node.metadata.project_slug).filter(slug => slug !== undefined)
+    ),
+  ];
+
+  const projectPlatforms = projectSlugs
+    .map(slug => projects.find(p => p.slug === slug))
+    .filter(project => project !== undefined)
+    .map(project => project?.platform ?? '');
```

Comment:
```suggestion
  const projectPlatforms = projects
    .filter(({slug}) => projectSlugs.includes(slug))
    .map(project => project?.platform ?? '');
```

Just a small suggestion, instead of nesting the `.find` inside the `.map` we could just filter outright for the projects and I think that eliminates the need to do an undefined check

Reply:
If the platform _is_ undefined (I do think there are data issues where this does happen, or somehow was able to happen) should we leave it as the empty string or should we give it a different name like 'unknown'? Whatever works for the intended analytics.

Reply:
thanks, updated the find/map. I'm keeping the project platforms empty because that's our standard behaviour on Amplitude for missing entries.